apps/api/src/rating/dto/create-rating.dto.ts

```diff
@@ -0,0 +1,23 @@
+import { Type } from 'class-transformer';
+import { IsNotEmpty, IsNumber, IsString, Max, Min } from 'class-validator';
+
+export class CreateRatingDto {
+  @IsNotEmpty()
+  readonly productId: string;
+
+  @IsNotEmpty()
+  @IsNumber()
+  @Min(1, {
+    message: 'Minimum rating is 1',
+  })
+  @Max(5, {
+    message: 'Maximum rating is 5',
+  })
```

Comment:
```suggestion
  @isValidRating()
```


add this to custom-validation.utils.ts
```
export function isValidRating(options?: ValidationOptions): PropertyDecorator {
  return function (object: Record<string, any>, propertyName: string) {
    registerDecorator({
      name: 'isValidRating',
      target: object.constructor,
      propertyName: propertyName,
      options: options,
      validator: {
        validate(value: any): boolean {
          return typeof value === 'number' && value >= 1 && value <= 5;
        },
        defaultMessage(): string {
          return 'Rating must be a number between 1 and 5.';
        },
      },
    });
  };
}
```